src/js/background/background.js

```diff
@@ -478,70 +488,79 @@ async function displayBrowserActionBadge() {
   }
 }
 
-browser.runtime.onMessage.addListener(async (m, sender, _sendResponse) => {
-  let response;
-
-  switch (m.method) {
-    case "displayBrowserActionBadge":
-      await displayBrowserActionBadge();
-      break;
-    case "iframeCloseRelayInPageMenu":
-      browser.tabs.sendMessage(sender.tab.id, {message: "iframeCloseRelayInPageMenu"});
-      break;
-    case "fillInputWithAlias":
-      browser.tabs.sendMessage(sender.tab.id, m.message);
-      break;
-    case "updateIframeHeight":
-      browser.tabs.sendMessage(sender.tab.id, m);
-      break;
-    case "getServerStoragePref":
-      response = await getServerStoragePref();
-      break;
-    case "getAliasesFromServer":
-      response = await getAliasesFromServer("GET", m.options);
-      break;
-    case "patchMaskInfo":
-      await patchMaskInfo("PATCH", m.id, m.data, m.options);
-      break;
-    case "getCurrentPageHostname":
-      // Only capture the page hostanme if the active tab is an non-internal (about:) page.
-      response = await getCurrentPageHostname();
-      break;
-    case "makeDomainAddress":
-      response = await makeDomainAddress(m.address, m.block_list_emails, m.description);
-      break;
-    case "makeRelayAddress":
-      response = await makeRelayAddress(m.description);
-      break;
-    case "postReportWebcompatIssue":
-      response = await postReportWebcompatIssue(m.description);
-      break;
-    case "openRelayHomepage":
-      browser.tabs.create({
-        url: `${RELAY_SITE_ORIGIN}?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=go-to-fx-relay`,
-      });
-      break;
-    case "rebuildContextMenuUpgrade":
-      await relayContextMenus.init();
-      break;
-    case "refreshAccountPages":
-      await refreshAccountPages();
-      break;
-    case "sendMetricsEvent":
-      response = await sendMetricsEvent(m.eventData);
-      break;
-    case "updateAddOnAuthStatus":
-      await updateAddOnAuthStatus(m.status);
-      break;
-    case "updateInputIconPref":
-      browser.storage.local.set({ showInputIcons: m.iconPref });
-      break;
-  }
-  return response;
-});
+browser.runtime.onMessage.addListener((m, sender, sendResponse) => {
```

Comment:
In MPP-3490, when it says "Ensure listeners are registered synchronously at the top-level", what exactly does this mean? Do all listeners need to be registered synchronously? If so there are many others that don't follow this format.

Reply:
If you look at what we [currently have](https://github.com/mozilla/fx-private-relay-add-on/blob/main/src/js/background/background.js#L489) on background.js, you'll see it's an async message. 

This same change is required for migrating to Manifest v3, so they're somewhat related.

Reply:
I see the three you mean: 
- `browser.contextMenus.onClicked.addListener`(`background.js`)
- `browser.menus.onShown.addListener` (`context-menu.js`)
- `browser.runtime.onInstalled.addListener` (`context-menu.js`)

Specifically,  the two items in context-menu don't apply as that code/logic is not ran on Firefox for Android, but the start-up def. is, and may be impacting the mobile first-run. 

Solid call out. I'll adjust the `browser.runtime.onInstalled.addListener` function accordingly.

Reply:
Oh in MPP-3490, by "top-level" did it mean the ones in background.js? I was also wondering about other listeners in different files.

Reply:
Yeah - Any `browser.foo` level listeners.